lib/archethic/db/embedded_impl/chain_writer.ex

```diff
@@ -9,16 +9,18 @@ defmodule Archethic.DB.EmbeddedImpl.ChainWriter do
   alias Archethic.TransactionChain.Transaction
   alias Archethic.TransactionChain.Transaction.ValidationStamp
 
-  def start_link(arg \\ []) do
-    GenServer.start_link(__MODULE__, arg, name: __MODULE__)
+  def start_link(arg \\ [], opts \\ []) do
+    GenServer.start_link(__MODULE__, arg, opts)
   end
 
   @doc """
   Append a transaction to a file for the given genesis address
   """
   @spec append_transaction(binary(), Transaction.t()) :: :ok
   def append_transaction(genesis_address, tx = %Transaction{}) do
-    GenServer.call(__MODULE__, {:append_tx, genesis_address, tx})
+    partition = :erlang.phash2(genesis_address, 20)
```

Comment:
?? phash2?

Reply:
It's a hash fn which returns a value between the range specified for a given input. This isparticularly useful when you want to create partitions.